### PR TITLE
Update whitelist and groovy whitelist to add inject and split method

### PR DESCRIPTION
**Pull request by castorls**
We are using this two methods to create docker parameters lines from environements variables (compatible with withEnv step).

The code we are using actually are present in the test file.

A snapshot version with this commit is already on my production server.

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist-groovy2 [hunk 1]
```diff
@@ -6,6 +6,15 @@ staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods each java.util.Lis
 staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods each java.util.Map groovy.lang.Closure
 staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods each java.util.Set groovy.lang.Closure
 staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods each java.util.SortedSet groovy.lang.Closure
+
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.lang.Iterable java.lang.Object groovy.lang.Closure
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.util.Collection java.lang.Object groovy.lang.Closure
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.util.Iterator java.lang.Object groovy.lang.Closure
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist-groovy2]: At least this overload, and probably others, is present in Groovy 1.x and thus should be listed in `generic-whitelist`, not here.

File: src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy [hunk 2]
```diff
@@ -26,6 +26,13 @@ assert arr == [0];
 assert true in [true, false];
 (1..3).each {};
 [1..3]*.toString();
+
+// convert env variable to docker env parameters
+assert "test=test -e test2=test2" == ["test=test","test2=test2"].inject([]) { result, entry ->
```
Comment (jglick) [inline src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy]: Patch to this file can just be reverted. This test case is only about syntactic forms. There is no need to have tests for every whitelisted method.

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist-groovy2 (hunk 1 above)
Comment (castorls) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist-groovy2]: OK, I migrate this definition to generic-whitelist file. Could you explain me what should contains this file ?

File: src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy (hunk 2 above)
Comment (castorls) [inline src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy]: OK, I thought that the added elements should be tested in this file. I revert it.

File: src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy [hunk 3]
```diff
@@ -26,6 +26,7 @@ assert arr == [0];
 assert true in [true, false];
 (1..3).each {};
 [1..3]*.toString();
+
```
Comment (jglick) [inline src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy]: gratuitous

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist [hunk 4]
```diff
@@ -213,6 +214,11 @@ staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods getAt java.util.Li
 staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods getAt java.util.Map java.lang.Object
 staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods getAt java.util.regex.Matcher int
 staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods getChars java.lang.String
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.lang.Object java.lang.Object groovy.lang.Closure
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.lang.Object[] java.lang.Object groovy.lang.Closure
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.util.Collection java.lang.Object groovy.lang.Closure
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.util.Iterator java.lang.Object groovy.lang.Closure
+staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods inject java.util.Map java.lang.Object groovy.lang.Closure
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist]: merge conflicts I think

File: pom.xml [hunk 5]
```diff
@@ -8,7 +8,7 @@
   </parent>
 
     <artifactId>script-security</artifactId>
-    <version>1.29-SNAPSHOT</version>
+    <version>1.28-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml]: revert

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist (hunk 4 above)
Comment (castorls) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist]: Yes I haven't view that you have released a new version.

File: src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy (hunk 3 above)
Comment (castorls) [inline src/test/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest/all.groovy]: effectively, I've forgotten to remove it as the previous commit

**Pull request merged**